atc/api/configserver/save.go

```diff
@@ -69,7 +69,22 @@ func (s *Server) SaveConfig(w http.ResponseWriter, r *http.Request) {
 	}
 
 	pipelineName := rata.Param(r, "pipeline_name")
+	err := atc.ValidateIdentifier(pipelineName, "pipeline")
+	if err != nil {
+		warnings = append(warnings, atc.ConfigWarning{
```

Comment:
Based on the RFC, we should be able to save existing pipelines with invalid names, but we should disallow creating new pipelines:

> To ease this pain, we can allow existing data to stay as-is, and **only enforce the identifier rules for newly created teams and pipelines**

So, maybe we should add a check to determine whether it's a new pipeline or not (i.e. if the pipeline exists yet or not) - if it is new, we error with a 400 Bad Request, and if not, we add the warning

Reply:
Actually, this makes me realize that we should be strict about all the identifier rules for newly created pipelines (all of the changes to `configvalidate` and `step_validator` as well). Not sure what the best way to do this is - the easiest thing I can think of is to special case on `invalid_identifier` warnings and return a 400 error if the pipeline is new. Open to any ideas you have!